cpu/kinetis_common/include/periph_cpu.h

```diff
@@ -51,6 +51,32 @@ typedef uint16_t gpio_t;
 #define CPUID_LEN           (16U)
 
 /**
+ * @brief   Generate GPIO mode bitfields
+ *
+ * We use the following bits to encode the pin mode:
+ * - bit 0: 0 for pull-down or 1 for pull-up
+ * - bit 1: pull register enable (as configured in bit 0)
+ * - bit 5: OD enable
+ * - bit 7: output or input mode
+ */
+#define GPIO_MODE(pu, pe, od, out)   (pu | (pe << 1) | (od << 5) | (out << 7))
```

Comment:
I have to say I am not a big fan of these `_PS_SHIFT` macros - using plain nubmers here is un-ambiguous, well documented, and IMHO easier to read than those macros.